src/shogun2-core/shogun2-service/src/main/java/de/terrestris/shogun2/service/GeoServerInterceptorService.java

```diff
@@ -0,0 +1,611 @@
+					"set in interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service and EndPoint set in " +
+					"interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", requestOperation);
+		filterMap.put("endPoint", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service and Operation set in " +
+					"interceptor rule");
+			return result.get(0);
+		}
+
+		filterMap.put("operation", null);
+		filterMap.put("endPoint", null);
+
+		result = this.interceptorRuleService.findSpecificRule(filterMap);
+
+		if (!result.isEmpty()) {
+			LOG.debug(ruleEvent + ": Found Service set in interceptor rule");
+			return result.get(0);
+		}
+
+		return null;
+
+	}
+
+	/**
+	 *
+	 * @param mutableRequest
+	 * @param key
+	 * @return
+	 * @throws InterceptorException
+	 */
+	private static String getRequestParameterValue(MutableHttpServletRequest mutableRequest,
+			String[] keys) throws InterceptorException {
+
+		String value = StringUtils.EMPTY;
+
+		for (String key : keys) {
+			value = getRequestParameterValue(mutableRequest, key);
+
+			if (StringUtils.isNoneEmpty(value)) {
+				break;
+			}
+		}
+
+		return value;
+	}
+
+	/**
+	 *
+	 * @param mutableRequest
+	 * @param key
+	 * @return
+	 * @throws InterceptorException
+	 */
+	private static String getRequestParameterValue(MutableHttpServletRequest mutableRequest,
+			String key) throws InterceptorException {
+
+		if (StringUtils.isEmpty(key)) {
+			throw new InterceptorException("Missing parameter key");
+		}
+
+		String value = StringUtils.EMPTY;
+
+		Map<String, String[]> queryParams = mutableRequest.getParameterMap();
+
+		if (!queryParams.isEmpty()) {
+
+			TreeMap<String, String[]> params = new TreeMap<String, String[]>(
+					String.CASE_INSENSITIVE_ORDER);
+
+			params.putAll(queryParams);
+
+			if (params.containsKey(key)) {
+				value = StringUtils.join(params.get(key), ";");
+			}
+
+		} else {
+
+			String xml = OgcXmlUtil.getRequestBody(mutableRequest);
+
+			if (!StringUtils.isEmpty(xml)) {
+
+				Document document = OgcXmlUtil.getDocumentFromString(xml);
+
+				if (key.equalsIgnoreCase(OgcNaming.PARAMETER_SERVICE)) {
+					value = OgcXmlUtil.getPathInDocument(
+							document, "/*/@service");
+				} else if (key.equalsIgnoreCase(OgcNaming.PARAMETER_OPERATION)) {
+					value = OgcXmlUtil.getPathInDocument(
+							document, "name(/*)");
+
+					if (value.contains(":")) {
+						value = value.split(":")[1];
+					}
+
+				} else if (Arrays.asList(OgcNaming.PARAMETER_ENDPOINT).contains(key)) {
+					value = OgcXmlUtil.getPathInDocument(
+							document, "//TypeName/text()");
+					if (StringUtils.isEmpty(value)) {
+						value = OgcXmlUtil.getPathInDocument(document,
+								"//@typeName");
+					}
```

Comment:
Looking forward seeing tests against this method. The XPath expressions aren't so easy to grasp